### PR TITLE
Connman

### PR DESCRIPTION
**Pull request by nvinson**

### CODE REVIEW

File: net-misc/connman/Manifest [hunk 1]
```diff
@@ -1,3 +1,2 @@
 DIST connman-1.29.tar.xz 653324 SHA256 2a5a69693566f7fd59b2e677fa89356ada6d709998aa665caef8707b1e7a8594 SHA512 19f623dd23c6312f29bbd570dcb22d31154ffcd16ee66ed2280df3d9d844301f18538418363222d99eabdc33da0021488ed66ba2e4f88df30646004b724538ee WHIRLPOOL 9439cadeb20b72c922b1426da213f8567fe7b0da45f9a4abf270c38687a9d9161a8ee14250623344864d612e130e6f28926918468aaf4346a9e5ff41e6ad22bb
-DIST connman-1.32.tar.xz 668404 SHA256 3185864c73206a6033d12e9f583689dcd03f714a40a58333709d3f74a4e0934c SHA512 031b876f98b27c1a657af1e5242d06d9295f3710053c365fc94b13f386fe25e150ef95fbed81462c0d770522625bd5f55c621f5ecb1ad638dd64779c3e444066 WHIRLPOOL b20652bcb19b172b09aa3c10de3736eaeb500bd56a131b8580d750bf72ef9ba034b80c563b929f9d81de226b381e56b2aeee84186db66b0d169ffd2463e9417b
```
Comment (mgorny) [inline net-misc/connman/Manifest]: You've got the Manifest update in the wrong commit.

File: net-misc/connman/connman-1.33-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	"${FILESDIR}/${PN}-1.31-xtables.patch"
+)
+
+src_prepare() {
+	local mf="Makefile.am"
+
+	# Fix for bug # 591596
+	awk '/tmpfilesdir = @SYSTEMD_TMPFILESDIR@/{getline;next}{print}' \
+		$mf > $mf.tmp || die "Failed to remove systemd tmpfilesdir blob"
+	mv $mf.tmp $mf || die "failed to rename $mf.tmp to $mf"
```
Comment (mgorny) [inline net-misc/connman/connman-1.33-r1.ebuild]: Isn't this large bulk equivalent to `sed -i -e '/tmpfilesdir =/d' Makefile.am`? Also, are you convinced you can remove a directory like that with no consequences to its uses?

Also, is the issue reported upstream? What course of action did upstream approve?

File: net-misc/connman/connman-1.33-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+	sed -i -e 's/ProtectSystem=full/ProtectSystem=true/' \
```
Comment (mgorny) [inline net-misc/connman/connman-1.33-r1.ebuild]: Likewise, did this go upstream? Add an explanation why you're doing that.

File: net-misc/connman/connman-1.33-r1.ebuild [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+PATCHES=(
+	"${FILESDIR}/${PN}-1.32-execinfo-assumptions.patch"
+	"${FILESDIR}/${PN}-1.31-xtables.patch"
+)
+
+src_prepare() {
+	local mf="Makefile.am"
+
+	# Fix for bug # 591596
```
Comment (mgorny) [inline net-misc/connman/connman-1.33-r1.ebuild]: Add a more verbose explanation. People don't want to load 3 bugs to figure out why substitutions are done.

File: net-misc/connman/connman-1.33-r1.ebuild [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+	awk '/tmpfilesdir = @SYSTEMD_TMPFILESDIR@/{getline;next}{print}' \
+		$mf > $mf.tmp || die "Failed to remove systemd tmpfilesdir blob"
+	mv $mf.tmp $mf || die "failed to rename $mf.tmp to $mf"
+	sed -i -e 's/ProtectSystem=full/ProtectSystem=true/' \
+		src/connman.service.in || die "failed to update connman.service.in"
+
+	# Fix for bug # 596276
+	sed -i -e '/actiondir/s/polkit/polkit-gobject-1/' configure.ac ||
```
Comment (mgorny) [inline net-misc/connman/connman-1.33-r1.ebuild]: Likewise.

File: net-misc/connman/connman-1.33-r1.ebuild [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+	default
+	eautoreconf
+}
+
+src_configure() {
+	econf \
+		--localstatedir=/var \
+		--with-systemdunitdir=$(systemd_get_systemunitdir) \
+		--with-tmpfilesdir=$(get_libdir)/tmpfiles.d \
```
Comment (mgorny) [inline net-misc/connman/connman-1.33-r1.ebuild]: What does this do? I don't recall autotools having any sane behavior for passing relative paths. Not to mention: didn't you just comment that directory out?

File: net-misc/connman/Manifest (hunk 1 above)
Comment (nvinson) [inline net-misc/connman/Manifest]: Interesting.  I wonder how I did that.  Fixed now.

File: net-misc/connman/connman-1.33-r1.ebuild (hunk 4 above)
Comment (nvinson) [inline net-misc/connman/connman-1.33-r1.ebuild]: comments and inline substitutions have been removed in favor of two new patches.

File: net-misc/connman/connman-1.33-r1.ebuild (hunk 3 above)
Comment (nvinson) [inline net-misc/connman/connman-1.33-r1.ebuild]: talked to #gentoo-systemd no longer making this change.

File: net-misc/connman/connman-1.33-r1.ebuild (hunk 6 above)
Comment (nvinson) [inline net-misc/connman/connman-1.33-r1.ebuild]: autotools doesn't.  I thought get_libdir() provided absolute paths.  I've changed that line.  The code in Makefile.am is no longer commented out.

File: net-misc/connman/connman-1.33-r1.ebuild (hunk 2 above)
Comment (nvinson) [inline net-misc/connman/connman-1.33-r1.ebuild]: identical to ```sed -i -e '/tmpfilesdir =/,+1d'```.  Has been removed in favor of an actual patch.

Patches will be sent to upstream for inclusion.

**Comment by dlan17** [comment]
@nvinson, I've accepted this commit and pushed them to tree, but one thing I'd say: I'd really like the "one commit per fix" rule, and hope you can improve next time

reason:
1) easy for review, we could better know which changes fix which bug, instead of one commit fix three bugs
2) easy for maintain, we could only revert one of them if something bad happen

**Comment by nvinson** [comment]
> @nvinson, I've accepted this commit and pushed them to tree, but one thing I'd say: I'd really like the "one commit per fix" rule, and hope you can improve next time

Sorry, but no.  The "one commit per fix" rule, if applied here, would either result in splitting a revision bump across 3 commits (violation of Gentoo policy), or require the creation of multiple ebuild revisions that, if done properly, would all have to be run-tested as each bug would require a revision bump (see Gentoo devmanual on when revision bumps are required).  These interim ebuilds would then be removed in the second commit where I cleaned up old ebuilds.  The end result is a lot of churn with little improvement in the quality of the commit.  If anything, such a change would probably just produce a lot of unnecessary noise in the PR making the review harder not easier.

Gentoo's policy is that each commit be one logical change while intentionally not defining what a 'logical change' is.  The rationale for not defining a 'logical change' is so you don't end up with a bunch of needless commits into the tree (see https://wiki.gentoo.org/wiki/Gentoo_git_workflow#Commit_policy).

Having said that, if the bugs in question did not individually or as a group require an ebuild revision bump, then the "logical change" would have been on the bug level and "one fix per commit" would be expected and required by policy.

Because the revision bump was not the primary goal of the commit, I chose not to focus on it in the commit message.  In the future, I will make sure to call out when my commits are performing a revision bump, so it's more clear that the "logical change" is on an ebuild level and not an individual commit level.

**Comment by idella** [comment]
@dlan17 when was it ever advised to split commits to a single revbump?
NO
Look at it. S single revert of the revbump will suffice in any case.
One fix or atomic commits suit changes to an already established and present ebuild.
and hope you can improve next time

**Pull request closed without merging**